IPC_computetrigger.py

```diff
@@ -0,0 +1,189 @@
+    df["threshold_ML2_3_30"] = df.apply(lambda x: get_trigger(x, "ML2", 3, 30), axis=1)
+    df["threshold_ML2_3_5i"] = df.apply(
+        lambda x: get_trigger_increase(x, "ML2", 3, 5), axis=1
+    )
+
+    df["trigger_ML1"] = (df["threshold_ML1_4_20"]) | (
+        (df["threshold_ML1_3_30"] == 1) & (df["threshold_ML1_3_5i"] == 1)
+    )
+    df["trigger_ML2"] = (df["threshold_ML2_4_20"]) | (
+        (df["threshold_ML2_3_30"] == 1) & (df["threshold_ML2_3_5i"] == 1)
+    )
+    return df
+
+
+def main(country_iso3, config_file="config.yml"):
+    # TODO: now admin_level is defined by the config, prob better to make it an input arg
+    parameters = parse_yaml(config_file)[country_iso3]
+    country = parameters["country_name"]
+    admin_level = parameters["admin_level"]
+    processed_fews_path = parameters["processed_fews_path"]
+    processed_globalipc_path = parameters["processed_globalipc_path"]
+
+    ipc_cols = [
+        f"{status}_{i}"
+        for status in ["CS", "ML1", "ML2"]
+        for i in [1, 2, 3, 4, 5, "3p", "2m"]
+    ] + [
+        f"perc_{status}_{i}"
+        for status in ["CS", "ML1", "ML2"]
+        for i in [1, 2, 3, 4, 5, "3p", "2m"]
+    ]
+    pop_cols = [
+        f"pop_{status}" for status in ["CS", "ML1", "ML2", f"ADMIN{admin_level}"]
+    ]
+
+    # initialize dataframes such that can later check if they are filled with data
+    df_fewss = None
+    df_gipcs = None
+
+    if os.path.exists(processed_fews_path):
+        df_fews = pd.read_csv(processed_fews_path, index_col=0)
+        # TODO: adjust column names in process_fewsnet.py instead
+        df_fews = df_fews.rename(
+            columns={
+                parameters["shp_adm1c"]: "ADMIN1",
+                "adjusted_population": f"pop_ADMIN{admin_level}",
+            }
+        )
+        df_fews = add_columns(df_fews, "FewsNet")
+        df_fewss = df_fews[
+            ["date", "Source", f"ADMIN{admin_level}"] + pop_cols + ipc_cols
+        ]
+
+    if os.path.exists(processed_globalipc_path):
+        df_gipc = pd.read_csv(processed_globalipc_path, index_col=0)
+        df_gipc = add_columns(df_gipc, "GlobalIPC")
+        df_gipcs = df_gipc[
+            ["date", "Source", f"ADMIN{admin_level}"] + pop_cols + ipc_cols
+        ]
+
+    if df_fewss is not None and df_gipcs is not None:
+        df_comb = pd.concat([df_fewss, df_gipcs])
+        df_comb_trig = compute_trigger(df_comb)
+
+    elif df_fewss is not None:
+        df_comb = df_fewss
+        df_comb_trig = compute_trigger(df_comb)
+        logger.warning("No Global IPC data found")
+
+    elif df_gipcs is not None:
+        df_comb = df_gipcs
+        df_comb_trig = compute_trigger(df_comb)
+        logger.warning("No FewsNet data found")
+
+    else:
+        df_comb_trig = pd.DataFrame()
+        logger.warning("No data found")
+
+    RESULT_FOLDER = f"{country}/Data/IPC_trigger/"
+    Path(RESULT_FOLDER).mkdir(parents=True, exist_ok=True)
```

Comment:
should mkdir be in this .py if it is intended to be run every time a forecast update is available?

Reply:
the "exist_ok=True" argument causes the dir only to be made when it doesn't exist yet so would say that is fine. What would you suggest?